apcupsd-influxdb-exporter.py

```diff
@@ -5,47 +5,61 @@
 from apcaccess import status as apc
 from influxdb import InfluxDBClient
 
-
 dbname = os.getenv('INFLUXDB_DATABASE', 'apcupsd')
 user = os.getenv('INFLUXDB_USER')
 password = os.getenv('INFLUXDB_PASSWORD')
 port = os.getenv('INFLUXDB_PORT', 8086)
-host = os.getenv('INFLUXDB_HOST', 'localhost')
-interval = float(os.getenv('INTERVAL', 5))
-client = InfluxDBClient(host, port, user, password, dbname)
+host = os.getenv('INFLUXDB_HOST')
+apcupsdHost = os.getenv('APCUPSD_HOST', host)
+
+delay = os.getenv('DELAY', 10)
+
+removeTheseKeys = ['DATE', 'STARTTIME', 'END APC']
+tagKeys = ['APC', 'HOSTNAME', 'UPSNAME', 'VERSION', 'CABLE', 'MODEL', 'UPSMODE', 'DRIVER', 'APCMODEL', 'END APC']
 
+wattsKey = 'WATTS'
+nominalPowerKey = 'NOMPOWER'
+
+client = InfluxDBClient(host, port, user, password, dbname)
 client.create_database(dbname)
 
 while True:
-    ups = apc.parse(apc.get(host=os.getenv('APCUPSD_HOST', 'localhost')), strip_units=True)
-
-    watts = float(os.getenv('WATTS', ups.get('NOMPOWER', 0.0))) * 0.01 * float(ups.get('LOADPCT', 0.0))
-    json_body = [
-        {
-            'measurement': 'apcaccess_status',
-            'fields': {
-                'WATTS': watts,
-                'STATUS': ups.get('STATUS'),
-                'LOADPCT': float(ups.get('LOADPCT', 0.0)),
-                'BCHARGE': float(ups.get('BCHARGE', 0.0)),
-                'TONBATT': float(ups.get('TONBATT', 0.0)),
-                'TIMELEFT': float(ups.get('TIMELEFT', 0.0)),
-                'NOMPOWER': float(ups.get('NOMPOWER', 0.0)),
-                'CUMONBATT': float(ups.get('CUMONBATT', 0.0)),
-                'BATTV': float(ups.get('BATTV', 0.0)),
-                'OUTPUTV': float(ups.get('OUTPUTV', 0.0)),
-                'ITEMP': float(ups.get('ITEMP', 0.0)),
-            },
-            'tags': {
-                'host': os.getenv('HOSTNAME', ups.get('HOSTNAME', 'apcupsd-influxdb-exporter')),
-                'serial': ups.get('SERIALNO', None),
+    try:
+        ups = apc.parse(apc.get(host=apcupsdHost), strip_units=True)
+        for key in removeTheseKeys:
+            del ups[key]
```

Comment:
There's technically a potential error here, because you're deleting from ups without checking to make sure the key exists. This StackOverflow article has a nice answer to use .pop() https://stackoverflow.com/questions/8995611/removing-multiple-keys-from-a-dictionary-safely